File: python/hello.py
# 1. HTTP GET example.com
# 2. Regex title out of HTML
# 3. Reverse string
# 4. Write to a text file.

import requests
import re

# 1. HTTP GET example.com
r = requests.get('http://example.com')

# 2. Regex title out of HTML
title = re.findall("<title>(.*?)</title>", r.text)[0]

# 3. Reverse string
title = title[::-1]

# 4. Write to a text file.
with open('output.txt','w') as f:
    f.write(title)
f.closed

File: perl/hello.pl
#! /usr/bin/perl

# 1. HTTP GET example.com
# 2. Regex title out of HTML
# 3. Reverse string
# 4. Write to a text file.

use strict;
use LWP::Simple;

# 1. HTTP GET example.com
my $response = get("http://example.com");

# 2. Regex title out of HTML
$response =~ m|<title>(.*)</title>|;
my $title = $1;

# 3. Reverse string
$title = reverse($title);

# 4. Write to a text file.
my $filename = 'output.txt';
open(my $fh, '>', $filename) or die "Could not open file '$filename' $!";
print $fh $title;
close $fh;
